modules/context/permission.go

```diff
@@ -29,6 +29,16 @@ func RequireRepoWriter(unitType unit.Type) func(ctx *Context) {
 	}
 }
 
+// CanEnableEditor checks if the user is allowed to write to the branch of the repo
+func CanEnableEditor() func(ctx *Context) {
+	return func(ctx *Context) {
+		if !ctx.Repo.Permission.CanWriteToBranch(ctx.Repo.BranchName) {
+			ctx.NotFound(ctx.Req.URL.RequestURI(), nil)
```

Comment:
This `ctx.NotFound(ctx.Req.URL.RequestURI(), nil)` seems to be copied&pasted from other code. 

But, it doesn't read good ...  the first parameter is for the error log, we have better write it as something like:

```
ctx.NotFound('CanWriteToBranch denies permission', nil)
```

Reply:
And all similar code could be updated, too.